lib/src/main/java/com/auth0/msg/Message.java

```diff
@@ -0,0 +1,108 @@
+package com.auth0.msg;
+
+import com.auth0.jwt.algorithms.Algorithm;
+
+import java.util.Map;
+
+/**
+ * This interface all the methods related to message processing.
+ */
+public interface Message {
```

Comment:
Please add `throws InvalidClaimsException` to all the to methods. I have explained the reason in the abstract class.